pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1468,23 +1468,29 @@ func (c *Cloud) AttachDisk(diskName string, instanceName string, readOnly bool)
 	// See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
 	ec2Device := "/dev/xvd" + string(mountDevice)
 
-	if !alreadyAttached {
-		request := &ec2.AttachVolumeInput{
-			Device:     aws.String(ec2Device),
-			InstanceId: aws.String(awsInstance.awsID),
-			VolumeId:   aws.String(disk.awsID),
-		}
+	request := &ec2.AttachVolumeInput{
+		Device:     aws.String(ec2Device),
+		InstanceId: aws.String(awsInstance.awsID),
+		VolumeId:   aws.String(disk.awsID),
+	}
 
-		attachResponse, err := c.ec2.AttachVolume(request)
-		if err != nil {
+	attachResponse, err := c.ec2.AttachVolume(request)
+	if err != nil {
```

Comment:
I checked the getMountDevice function, the possible returns are
1. volume is already attached and assigned a device --> return values (device, alreadyAttached=true, nil) 
2. volume is attaching (?) and this function will try to assign an unused device to the volume --> return values (assignedDevice, alreadyAttached=false, nil)
3. volume is attaching but could not find any unused device to assgin --> ("", alreadyAttached=false, ERROR("too many volumes")
4. volume is not attached --> ("", alreadyAttached=false, nil)
5. could not get awsinstance or describe instance --> ("", false, ERROR)

For my understanding, in 
case 1 and 2: this AttachDisk function can directly return device without error (case 2 assigns unused device to the volume)
case 3: return error since currently too many volumes assigned and could not find unused device
case 4: trigger Attach request
case 5: return error

But I might misunderstand this code totally...